airflow/providers/google/cloud/hooks/dataplex.py

```diff
@@ -361,3 +390,470 @@ def get_lake(
+        Gets a DataScan resource.
+
+        :param project_id: Required. The ID of the Google Cloud project that the lake belongs to.
+        :param region: Required. The ID of the Google Cloud region that the lake belongs to.
+        :param data_scan_id: Required. Data Quality scan identifier.
+        :param retry: A retry object used  to retry requests. If `None` is specified, requests
+            will not be retried.
+        :param timeout: The amount of time, in seconds, to wait for the request to complete.
+            Note that if `retry` is specified, the timeout applies to each individual attempt.
+        :param metadata: Additional metadata that is provided to the method.
+        """
+        client = self.get_dataplex_data_scan_client()
+
+        name = PATH_DATA_SCAN.format(project_id=project_id, region=region, data_scan_id=data_scan_id)
+        result = client.get_data_scan(
+            request={"name": name, "view": "FULL"},
+            retry=retry,
+            timeout=timeout,
+            metadata=metadata,
+        )
+        return result
+
+    @GoogleBaseHook.fallback_to_default_project_id
+    def update_data_scan(
+        self,
+        project_id: str,
+        region: str,
+        data_scan_id: str,
+        body: dict[str, Any] | DataScan,
+        update_mask: dict | FieldMask | None = None,
+        retry: Retry | _MethodDefault = DEFAULT,
+        timeout: float | None = None,
+        metadata: Sequence[tuple[str, str]] = (),
+    ) -> Any:
+        """
+        Updates a DataScan resource.
+
+        :param project_id: Required. The ID of the Google Cloud project that the lake belongs to.
+        :param region: Required. The ID of the Google Cloud region that the lake belongs to.
+        :param data_scan_id: Required. Data Quality scan identifier.
+        :param body: Required. The Request body contains an instance of DataScan.
+        :param update_mask: Required. Mask of fields to update.
+        :param retry: A retry object used  to retry requests. If `None` is specified, requests
+            will not be retried.
+        :param timeout: The amount of time, in seconds, to wait for the request to complete.
+            Note that if `retry` is specified, the timeout applies to each individual attempt.
+        :param metadata: Additional metadata that is provided to the method.
+        """
+        client = self.get_dataplex_data_scan_client()
+
+        full_scan_name = f"projects/{project_id}/locations/{region}/dataScans/{data_scan_id}"
+
+        if body:
+            if isinstance(body, DataScan):
+                body.name = full_scan_name
+            elif isinstance(body, dict):
+                body["name"] = full_scan_name
+            else:
+                raise AirflowException("Unable to set scan_name.")
+
+        if not update_mask:
+            update_mask = FieldMask(
+                paths=["data", "data_quality_spec", "labels", "description", "displayName", "executionSpec"]
```

Comment:
displayName --> display_name
executionSpec --> execution_spec

We should use snake_case for the paths in the field mask.

Reply:
changed, 
thanks for catching that